### PR TITLE
MaxBatchSize E2E Test 

### PR DESCRIPTION
**Pull request by pengwa**
**Description**: To avoid regression on the maximum batch size we can train BERT_LARGE leveraging ORT, we need an end to end testing.

**Motivation and Context**
- Why is this change required? What problem does it solve?
  Make sure we can find the memory regression for training at the earliest convenience. 
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by pengwa** [comment]
@edgchen1 Could you please help upload bert large into the zip on Azure? Or you can mail me steps to do that. Thanks a lot! 

> ARCHIVE_BLOB_URL = "https://onnxruntimetestdata.blob.core.windows.net/training/onnxruntime_training_data.zip?snapshot=2019-12-12T18:53:12.3943496Z"
> ARCHIVE_SHA256_DIGEST = "066abe039a544d4f100ad2047f5983b65dc27d0b24f9f448cf4474076b486500"

File: orttraining/tools/ci_test/run_batch_size_test.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import argparse
+import subprocess
+import sys
+import os
+
+SCRIPT_DIR = os.path.realpath(os.path.dirname(__file__))
```
Comment (edgchen1) [inline orttraining/tools/ci_test/run_batch_size_test.py:10]: looks like this isn't needed

File: cmake/onnxruntime_training_e2e_tests.cmake [hunk 2]
```diff
@@ -24,3 +24,18 @@ set_property(
...
+  NAME onnxruntime_training_bert_batch_size_test
+  COMMAND
+    ${Python3_EXECUTABLE} ${REPO_ROOT}/orttraining/tools/ci_test/run_batch_size_test.py
+      --binary_dir $<TARGET_FILE_DIR:onnxruntime_training_bert>
+      --model_root ${onnxruntime_TRAINING_E2E_TEST_DATA_ROOT}/models
+  CONFIGURATIONS RelWithDebInfo)
+
+set_property(
+  TEST
+    onnxruntime_training_bert_batch_size_test
```
Comment (edgchen1) [inline cmake/onnxruntime_training_e2e_tests.cmake:39]: consider using a single set_property() call:
```
set_property(
  TEST
    onnxruntime_training_bert_batch_size_test
    onnxruntime_training_bert_convergence_e2e_test
  PROPERTY
    LABELS training_e2e)
```

File: orttraining/tools/ci_test/run_batch_size_test.py (hunk 1 above)
Comment (pengwa) [reply orttraining/tools/ci_test/run_batch_size_test.py:10]: you are right!!! thanks man!

**Comment by pengwa** [comment]
@edgchen1  how to validate the e2e, manual trigger it?

**Comment by SherlockNoMad** [comment]
> @edgchen1 how to validate the e2e, manual trigger it?

Yes. Need to manually trigger it.

**Comment by pengwa** [comment]
Thank you @edgchen1, @SherlockNoMad

**Pull request merged**